datacenter/dtr/2.2/guides/admin/configure/jobrunner.md

```diff
@@ -0,0 +1,261 @@
+---
+title: Jobrunner
+description: Learn about the inner-workings of the jobrunner container in the DTR workflow.
+keywords: docker, job, runner
+---
+
+The jobrunner container is a DTR mechanism that:
+
+1. Consumes jobs from a cluster-wide jobs queue 
+2. Performs the work of the given action
+
+There is one jobrunner container per replica. 
+
+## Jobrunner Workflow
+The following diagram depicts the behavior of the jobrunner:
+
+[//]: # (Placeholder for jobrunner diagram. @sarahpark will work on the diagram)
+
+When a job is scheduled (see [Job Actions](#job-actions) below) it is put onto the 
+cluster-wide jobs queue with an initial status of `waiting`. When a jobrunner worker 
+is available to pick up the job, it will claim the it (i.e: the workerID will be set 
+to the replicaID of the jobrunner container that claimed the job) and set the job 
+status to `running`. The worker will carry out the job and then set the appropriate 
+status when complete (see [Job Statuses](#job-statuses) below).
+
+[//]: # (Placeholder for jobrunner scheduling. @sarahpark will work on the diagram)
+
+Each jobrunner has an internal queue of `waiting` jobs sorted by their `scheduledAt`
+time (from earliest to latest). When a worker is free to claim the next job, it claims
+it after a delay of up to 3 seconds. This delay is imposed so that each available worker
+has a chance to compete for the job. The worker that was successfully able to claim the
+job will set the job's `workerID` and all other workers will drop the job from their
+internal queue. If a job cannot be claimed due to capacity limits (see [Job Capacities](#job-capacities))
+then it is placed into a separate queue and will go through the claiming process
+when the worker has enough free capacity for it.
+
+Jobrunners monitor each other's `heartbeatExpiration` in the workers table. When a worker 
+see that another worker hasn't updated its expiration in a long time, it sets the 
+dead worker's status to `dead` and its jobs to `worker_dead`. If the dead worker is able to
+reconnect to the database and notices that it's jobs have been set to `worker_dead`,
+it sets those job statuses to `worker_resurrection` and cancels them.
+
+### Job Actions
+The available job actions are:
+
+- `gc`
+: Garbage collection deletes layers associated with deleted images.
+- `sleep`
+: Sleep is used to test the correctness of the jobrunner. It sleeps for 60 seconds.
+- `false`
+: False is used to test the correctness of the jobrunner. It runs the `false` command and immediately fails.
+- `tagmigration`
+- `bloblinkmigration`
```

Comment:
`bloblinkmigration` is a 2.1 to 2.1 upgrade process that adds references for blobs to repositories in the database